.github/workflows/containerdeploy.yml

```diff
@@ -30,144 +33,99 @@ jobs:
       - name: Downcase REPO
         run: echo "repo=${GITHUB_REPOSITORY,,}" >> $GITHUB_OUTPUT
         id: downcase_repo
-      - name: Checkout repository because we need to build contrib
-        uses: actions/checkout@v3
-        with:
-          submodules: true
-        if: startsWith(steps.extract_branch.outputs.branch, 'release/')
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
       - name: Login to GitHub Container Registry
-        uses: docker/login-action@v2
+        uses: docker/login-action@v3
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}        
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Checkout contrib submodule repository
+        run: git submodule update --init contrib
+        # we only need the contrib submodule if we're on a release branch
+        # otherwise, we use the latest contrib image built in the OpenMS/contrib
+        if: startsWith(steps.extract_branch.outputs.branch, 'release/')
       - name: Build contrib container using the correct submodule hash
-        uses: docker/build-push-action@v3
+        uses: docker/build-push-action@v5
         with:
           push: true
           context: contrib
           file: contrib/dockerfiles/Dockerfile
           tags: |
             ghcr.io/openms/contrib:${{ steps.tag_name.outputs.tag }}
-        if: startsWith(steps.extract_branch.outputs.branch, 'release/')        
-      - name: Checkout Library and Executable Dockerfiles
-        shell: bash # uses git bash on windows
-        run: |
-            git clone https://github.com/OpenMS/dockerfiles
-            cd dockerfiles
-            ## for release, try to checkout the specific release branch in the dockerfiles repo first
-            if [[ "${{ steps.extract_branch.outputs.branch }}" == "release/"* ]]
-            then
-              git checkout ${{ steps.extract_branch.outputs.branch }} || git checkout master
-            fi
+          cache-from: type=gha,scope=ghcr.io/openms/contrib:${{ steps.tag_name.outputs.tag }}
+          cache-to: type=gha,mode=max,scope=ghcr.io/openms/contrib:${{ steps.tag_name.outputs.tag }}
+        if: startsWith(steps.extract_branch.outputs.branch, 'release/')
       - name: Build and Push Docker Library Image
-        uses: docker/build-push-action@v3
+        uses: docker/build-push-action@v5
         with:
           push: true # Will only build if this is not here
-          context: dockerfiles/library
+          file: dockerfiles/Dockerfile
+          target: library
           build-args: |
-            OPENMS_TAG=latest
-            OPENMS_BRANCH=${{ steps.extract_branch.outputs.branch }}
+            NUM_BUILD_CORES=2
```

Comment:
I think GitHub recently upgraded their free runners. You might try 4.

Reply:
Nice! You're right: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories